src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
 // write code here
+
+const employees = document.querySelectorAll('li');
+const listEmployees = document.querySelector('ul');
+
+function getSalary(element) {
+  const salary = element.dataset.salary;
+
+  if (!salary) {
+    throw new Error(`Salary not found for employee: ${element.textContent}`);
+  }
+
+  const cleanedSalary = salary.substring(1).replace(',', '');
+  const numericSalary = Number(cleanedSalary);
+
+  if (isNaN(numericSalary)) {
+    throw new Error(`Invalid salary format for employee:
+      ${element.textContent}`);
+  }
+
+  return numericSalary;
+}
+
+function sortList(list) {
+  const listItem = Array.from(list);
+
+  const sortedList = listItem.slice().sort((a, b) => {
+    return getSalary(b) - getSalary(a);
+  });
+
+  listEmployees.innerHTML = '';
```

Comment:
Instead of setting `innerHTML` to an empty string to clear the list, you can use the `textContent` property. This is a safer option as it does not interpret the passed content as HTML, but only as text. So, it helps to prevent Cross-Site Scripting (XSS) attacks.